src/components/semantic/props/CheckboxDocProp.tsx

```diff
@@ -9,18 +9,31 @@ type CheckboxDocProps<K extends keyof D, D> =
     prop: K;
     label: string;
     checkedIfUndefined?: boolean;
+    disabled?: boolean;
 };
 
 export function CheckboxDocProp<K extends keyof D, D extends { [Key in K]?: boolean }>({
     doc,
     update,
     prop,
     label,
-    checkedIfUndefined
+    checkedIfUndefined,
+    disabled,
 }: CheckboxDocProps<K, D>) {
-    return <Label className={styles.checkboxLabel}>
+    useEffect(()=> {
+        if (disabled && doc[prop]) {
+            update({
+                ...doc,
+                [prop]: false,
+            });
+        }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [disabled, doc[prop]]);
+
+    return <Label className={styles.checkboxLabel} style={{color: disabled ? "gray" : "black"}}>
         <Input type="checkbox"
-               checked={doc[prop] ?? checkedIfUndefined ?? false}
+               disabled={disabled}
+               checked={(!disabled && doc[prop]) ?? checkedIfUndefined ?? false}
```

Comment:
Not sure you need this change. The `useEffect` should manage it for you!
```suggestion
               checked={doc[prop] ?? checkedIfUndefined ?? false}
```

Reply:
All of these changes seem reasonable and I'll add them in 😊. I think I'd been trying various things to stop `useEffect` from causing a re-render loop, which lead to some redundancies like this.